src/main/java/co/cask/gcs/sink/GCSBatchSink.java

```diff
@@ -125,10 +125,15 @@ public GCSBatchSinkConfig() {
 
     public void validate() {
       // validate project id availability
-      GCPUtil.getProjectId(project);
-      if (path != null && !path.startsWith("gs://")) {
+      if (!containsMacro("project")) {
+        GCPUtil.getProjectId(project);
+      }
```

Comment:
just realized, for other plugins we decided to validate projectId only during `prepareRun` and not during `config` as users could prefer to not provide project for CDAP local sandbox but pipelines configured to run on data-proc where its picked up automatically.

Reply:
good catch, will remove this check